src/esp/sensor/FisheyeSensor.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "FisheyeSensor.h"
+
+#include <Corrade/Utility/Assert.h>
+#include <Corrade/Utility/FormatStl.h>
+#include "esp/gfx/DoubleSphereCameraShader.h"
+
+namespace Mn = Magnum;
+namespace Cr = Corrade;
+
+namespace esp {
+namespace sensor {
+
+FisheyeSensorSpec::FisheyeSensorSpec() : VisualSensorSpec() {
+  uuid = "fisheye";
+  sensorSubType = SensorSubType::Fisheye;
+}
+
+void FisheyeSensorSpec::sanityCheck() {
+  VisualSensorSpec::sanityCheck();
+  CORRADE_ASSERT(
+      sensorSubType == SensorSubType::Fisheye,
+      "FisheyeSensorSpec::sanityCheck(): sensorSpec is not Fisheye", );
+
+  CORRADE_ASSERT(focalLength[0] > 0 && focalLength[1] > 0,
+                 "FisheyeSensorSpec::sanityCheck(): focal length,"
+                     << focalLength << "is illegal.", );
+}
+
+void FisheyeSensorDoubleSphereSpec::sanityCheck() {
+  FisheyeSensorSpec::sanityCheck();
+  CORRADE_ASSERT(fisheyeModelType == FisheyeSensorModelType::DoubleSphere,
+                 "FisheyeSensorDoubleSphereSpec::sanityCheck(): fisheye model "
+                 "type is wrong. It should be DoubleSphere.", );
+  CORRADE_ASSERT(alpha >= 0 && alpha <= 1.0,
+                 "FisheyeSensorDoubleSphereSpec::sanityCheck(): alpha"
+                     << alpha << "is illegal.", );
```

Comment:
Should any of these be `ESP_CHECK()` instead? AFAICT the alpha or focal length could be settable from python and thus a great candidate for a python exception.

Reply:
I do not have preference, either way is OK.
Yes, we can try it.

Reply:
Depends if you're going to expose those to python or not. If not, doesn't matter :)

Reply:
No, these functions will not be exposed directly to python. They will only be called in the sensor's constructor.